core/src/main/resources/schema/app/20210831072928_DISTX-552_Supporting_custom_service_configurations_while_launching_Data_Hub_clusters.sql

```diff
@@ -0,0 +1,49 @@
+-- // DISTX-552 Supporting custom service configurations while launching Data Hub clusters
+-- Migration SQL that makes the change goes here.
+CREATE SEQUENCE IF NOT EXISTS custom_configs_id_seq;
+
+CREATE TABLE IF NOT EXISTS customconfigs
+(
+    id                bigint DEFAULT nextval('custom_configs_id_seq'::regclass)                   NOT NULL
+                      CONSTRAINT customconfigs_pkey
+                      PRIMARY KEY,
+    name              VARCHAR(255)                                                                NOT NULL,
+    crn               VARCHAR(255)                                                                NOT NULL,
+    created           bigint DEFAULT (date_part('epoch'::text, now()) * (1000)::double precision) NOT NULL,
+    runtimeversion    VARCHAR(255),
+    account           VARCHAR(255)                                                                NOT NULL
+);
+
+CREATE UNIQUE INDEX IF NOT EXISTS idx_customconfigs_crn
+    ON customconfigs (crn);
+
+CREATE UNIQUE INDEX IF NOT EXISTS idx_customconfigs_name_acoount
+    ON customconfigs (name, account);
+
+CREATE SEQUENCE IF NOT EXISTS custom_config_property_id_seq;
+
+CREATE TABLE IF NOT EXISTS customconfigs_properties
+(
+    id               bigint DEFAULT nextval('custom_config_property_id_seq'::regclass) NOT NULL
+                     CONSTRAINT customconfigproperty_pkey
+                     PRIMARY KEY,
+    name             TEXT                                                              NOT NULL,
+    value            TEXT                                                              NOT NULL,
+    roletype         TEXT,
+    servicetype      TEXT                                                              NOT NULL,
+    customconfigs_id bigint
+                     CONSTRAINT fk_customconfig_property_customconfigs_id
+                     REFERENCES customconfigs
+                     ON UPDATE CASCADE ON DELETE CASCADE
+);
+
+ALTER TABLE cluster ADD COLUMN IF NOT EXISTS customconfigurationscrn varchar(255);
```

Comment:
at database level we should use the other table's primary key if it's possible, and define it as a foreign key, to enforce it exists in the other table. Is there any reason to not to do so in this case?

Reply:
Updated the schema as well as the entity to use foreign key